tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -3479,7 +3488,13 @@ def _to_proto(self):
     pb.optimization_options.CopyFrom(self.experimental_optimization._to_proto())  # pylint: disable=protected-access
     if self.experimental_slack is not None:
       pb.slack = self.experimental_slack
-    pb.threading_options.CopyFrom(self.experimental_threading._to_proto())  # pylint: disable=protected-access
+    # (kvignesh1420): We try to keep the values of `threading` and
+    # `experimental_threading` the same, to prevent unexpected behaviours.
+    if not self.threading._has_default_values():
+      self.experimental_threading = self.threading
```

Comment:
When you do this, you should check whether `experimental_threading` has any values that are different from `threading` and if so, log a warning with information about which `experimental_threading` options are being ignored.

Reply:
addressed in the commit

Reply:
addressed in the commit